src/services/BookingService.ts

```diff
@@ -46,14 +51,39 @@ class BookingService {
 
 
     if (bookingsAmount !== 0) {
-      throw new HttpException();
+      throw new HttpException(
+        500, "Slot is already occupied.",
+      );
     }
 
 
     return false;
   }
 
+  public async doesTheBookingOfTheUserOverlap(
+    userId: number,
+    startDate: string,
+    endDate: string,
+  ): Promise<boolean> {
+    const repo = this.getRepository();
+
+    const userBookings = await repo.findByIdAndDates(
+      {
+        entity  : "user",
+        entityId: userId,
+      },
+      startDate,
+      endDate,
+    );
+
+    if (userBookings.length !== 0) {
+      throw new HttpException(
+        500, "You already booked a slot in this time frame. Cancel it before booking a new slot.",
```

Comment:
400

Reply:
Make http errors more generic, you don't need to think on this as a end-user facing error. This is going to be displayed to users of the API (usually internal developers) and a proper error will be shown to end users. You can just say `User already has a booking slot in this time frame` without any potential solution to the problem.